internal/policy/verify.go

```diff
@@ -530,105 +502,83 @@ func verifyEntry(ctx context.Context, repo *git.Repository, policy *State, attes
-
-	// 3. Use each trusted key to verify signature
+	// Use each verifier to verify signature
 	rslEntryVerified := false
-	for _, key := range trustedKeys {
-		err := gitinterface.VerifyCommitSignature(ctx, commitObj, key)
+	for _, verifier := range verifiers {
+		err := verifier.Verify(ctx, entry.ID, authorizationAttestation)
 		if err == nil {
 			// Signature verification succeeded
 			rslEntryVerified = true
 			break
-		}
-		if errors.Is(err, gitinterface.ErrUnknownSigningMethod) {
-			// We encounter this for key types that can be used for gittuf
-			// policy metadata but not Git objects
-			continue
-		}
-		if !errors.Is(err, gitinterface.ErrIncorrectVerificationKey) {
+		} else if !errors.Is(err, ErrVerifierConditionsUnmet) {
 			// Unexpected error
 			return err
 		}
-		// Haven't found a valid key, continue with next key
+		// Haven't found a valid verifier, continue with next key
 	}
 
 	if !rslEntryVerified {
 		return fmt.Errorf("verifying RSL entry failed, %w", ErrUnauthorizedSignature)
 	}
 
-	// 4. Verify tag object
+	// Verify tag object
 	tagObjVerified := false
-	tagObj, err := gitinterface.GetTag(repo, entry.TargetID)
-	if err != nil {
-		// Likely indicates the ref is not pointing to a tag object
-		// What about lightweight tags?
-		return err
-	}
-
-	entryTagRef, err := repo.Reference(plumbing.ReferenceName(entry.RefName), true)
-	if err != nil {
-		return err
-	}
-
-	if entry.TargetID != entryTagRef.Hash() && entry.TargetID != tagObj.Target {
-		return fmt.Errorf("verifying RSL entry failed, tag reference set to unexpected target")
-	}
-
-	if len(tagObj.PGPSignature) == 0 {
-		return fmt.Errorf(noSignatureMessage)
-	}
-
-	for _, key := range trustedKeys {
-		err := gitinterface.VerifyTagSignature(ctx, tagObj, key)
+	for _, verifier := range verifiers {
+		// explicitly not looking at the attestation
+		// that applies to the _push_
+		err := verifier.Verify(ctx, entry.TargetID, nil)
 		if err == nil {
 			// Signature verification succeeded
 			tagObjVerified = true
 			break
-		}
-		if errors.Is(err, gitinterface.ErrUnknownSigningMethod) {
-			// We encounter this for key types that can be used for gittuf
-			// policy metadata but not Git objects
-			continue
-		}
-		if !errors.Is(err, gitinterface.ErrIncorrectVerificationKey) {
+		} else if !errors.Is(err, ErrVerifierConditionsUnmet) {
 			// Unexpected error
 			return err
 		}
-		// Haven't found a valid key, continue with next key
+		// Haven't found a valid verifier, continue with next key
 	}
 
 	if !tagObjVerified {
 		return fmt.Errorf("verifying tag object's signature failed, %w", ErrUnauthorizedSignature)
 	}
 
+	entryTagRef, err := repo.GetReference(entry.RefName)
+	if err != nil {
+		return err
+	}
+
+	tagTargetID, err := repo.GetTagTarget(entry.TargetID)
+	if err != nil {
+		return err
+	}
+
+	if entry.TargetID != entryTagRef && entry.TargetID != tagTargetID {
+		return fmt.Errorf("verifying RSL entry failed, tag reference set to unexpected target")
+	}
```

Comment:
nit: Would it make sense to check these first before potentially more expensive crypto operations? Or do we need to verify the signatures first before we can trust these values?